src/@core/services/login/ValidateFormService.ts

```diff
@@ -0,0 +1,34 @@
+import { LoginFormProps } from '@components/login/AppFormLogin'
+import { formatZodError } from '@core/utils/formatError'
+import { z, ZodError } from 'zod'
+
+// Agrega-se todos os dados do formulário junto com o atributo isError
+type LoginFormPropsReturn = LoginFormProps & { isError?: boolean }
```

Comment:
Apenas acrescentei mais um campo no formulário chamado ```isError``` pra não precisar disparar nenhum erro e facilitar a analise dentro do componente. Então se ```isError``` for true, todos os campos do formulário vão obrigatoriamente ser convertidos em string por ```formatZodError``` e seus valores vão ser a mensagem de erro definida no esquema do zod